src/EasyCaching.Redis/DefaultRedisCachingProvider.Async.cs

```diff
@@ -295,6 +295,56 @@ public override async Task BaseSetAllAsync<T>(IDictionary<string, T> values, Tim
 
             return result;
         }
+        
+        /// <summary>
+        /// Gets all async.
+        /// </summary>
+        /// <returns>The all async.</returns>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <typeparam name="T">The 1st type parameter.</typeparam>
+        public override async Task<IDictionary<string, CacheValue<T>>> BaseGetAllAsync<T>(CancellationToken cancellationToken = default)
+        {
+            var connection = _cache.Multiplexer;
+
+            var keyArray = connection?.GetServer(connection?.GetEndPoints().FirstOrDefault())
+                ?.Keys(database: _cache.Database, pattern: "*")?.ToArray();
+
+            if (keyArray?.Length == 0 || keyArray is null)
+                return new Dictionary<string, CacheValue<T>>();
+
+            var values = _cache.StringGet(keyArray?.Select(k => (RedisKey)k).ToArray());
```

Comment:
Here should consider the key prefix.

Reply:
I refactor this method with optional prefix.